CanDIG/katsu
# DIG-1418: fix enum and validator not in nested schema

## What's new
- Resolved bug: DonorWithClinicalData schema lacked enums and validators due to oversight in handling multiple schema versions.
- Implemented a base schema with all validators and enums.
- Updated DonorWithClinicalData to inherit correctly from the base schema.
- Updated schema JSON.

### Testing
1. Checkout the branch `sonchau/fix_nested_schema_enums`.
2. Run `make build-katsu` and `make compose-katsu`.
3. Execute `make test-integration`.
4. Verify schema.json to ensure DonorWithClinicalData includes enums and validators.
